Clock/AnalogClock.qml

```diff
@@ -53,10 +51,7 @@ Image {
         }
     }
 
-    Timer { //l'ora si aggiorna ogni 0.5 secondi
-        interval: 500; running: true; repeat: true
-        onTriggered: { currentDate=new Date() }
-    }
```

Comment:
Hai eliminato questo timer per usare `currentDate` di `Clock.qml` anche dentro i _figli_ `DigitalClock` e `AnalogClock`, ho capito bene?
Ecco avere un riferimento ad una variabile di un altro componente sparsa dentro il file non è bellissimo. Rende un po' più complicato il debugging. In generale è meglio avere un'interfaccia pulita e chiara per un componente.
Se `AnalogClock` e `DigitalClock` hanno bisogno di una `currentDate`, potresti metterla come property in cima al componente, ed eventualmente modificarla da fuori. Ricordati che sono binding, quindi sono vivi.
```
Item {
  id: analogClock
  property var currentDate
  ...
}
```
```
//Clock.qml
...
AnalogClock {
    currentDate: root.currentDate  // <- si aggiorna automaticamente al cambiare di root.currentDate
}
```